cartridges/int_klaviyo/cartridge/scripts/klaviyoATC.js

```diff
@@ -0,0 +1,63 @@
+'use strict';
+
+/* Script Modules */
+var app = require('*/cartridge/scripts/app');
+var recreateHelpers = require('*/cartridge/scripts/recreateCartHelpers');
+
+/* API Includes */
+var Logger = require('dw/system/Logger');
+
+
+function addProductToCart(decodedItems, cartObj) {
+    try {
+        var productList = decodedItems.length ? decodedItems : null;
+        var cart = cartObj;
+
+        if (cart.object.allProductLineItems.length) {
+            recreateHelpers.clearCart(cart.object)
+        }
+
+        var params = request.httpParameterMap;
+        var format = params.hasOwnProperty('format') && params.format.stringValue ? params.format.stringValue.toLowerCase() : '';
+        var newBonusDiscountLineItem;
+        var Product = app.getModel('Product');
+        var productOptionModel;
+        var productToAdd;
+        var template = 'checkout/cart/minicart';
+
+        if (params.source && params.source.stringValue === 'wishlist' && params.cartAction && params.cartAction.stringValue === 'update') {
+            app.getController('Wishlist').ReplaceProductListItem();
+            return;
+        } else {
+            var previousBonusDiscountLineItems = cart.getBonusDiscountLineItems();
```

Comment:
Is this logic for bonus discount line items needed? If we've emptied the cart already, there won't be any existing bonus product line items to process. It seems like the use case for this function is very strictly for Klaviyo cart recreate, unless I'm missing something. Any additional "add to cart" logic can probably be stripped out - this file has a unique name and doesn't appear to be an override to existing add to cart. I don't have the full context, so just a thought.

Reply:
Thanks for this - that line was part of the original  "add to cart" logic i kept in place. I've stripped that out as suggested and have tested with various encoded objects - this is functioning without these lines. Removed. Thanks again!